### PR TITLE
feature/35 added logout function

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

**Review by theblockstalk — COMMENTED** [review]
Please test logout in the ExternalUser.test.ts test

**Comment by sadiabbasi** [comment]
> ExternalUser

File: src/api/externalUser.ts [hunk 1]
```diff
@@ -64,8 +76,8 @@ export class ExternalUser {
             const accountName = await user.getAccountName();
 
             if (!accountName) {
-                //TODO: logout
-                // keyManager.clear(); must be implemented in future keymanager
+                console.log("logoutt")
```
Comment (theblockstalk) [inline src/api/externalUser.ts:79]: remove console log

File: test-integration/util/externalUser.ts [hunk 2]
```diff
@@ -193,3 +193,12 @@ export async function externalWebsiteOnReload(
...
 }
+
+
+export async function externalWebsiteOnLogout(
+    keyManager: KeyManager,
+    storageFactory: StorageFactory,
+) {
+    const externalUser = await ExternalUser.getUser({ keyManager, storageFactory });
+    await externalUser.logout();
+}
```
Comment (theblockstalk) [inline test-integration/util/externalUser.ts:204]: can you add some expect statements here to check storage?

**Review by theblockstalk — APPROVED** [review]

File: test-integration/externalUser.test.ts [hunk 3]
```diff
@@ -214,15 +215,16 @@ describe('External User class', () => {
 
             expect(TONOMY_LOGIN_WEBSITE_requests.length).toBe(2);
             expect(payload.accountName).toBe(await (await TONOMY_ID_user.getAccountName()).toString());
+            expect(payload.username).toBe(await (await TONOMY_ID_user.getUsername()).username?.split('.')[0]);
```
Comment (theblockstalk) [inline test-integration/externalUser.test.ts:218]: shouldnt this use `getBaseUsername()`?

Comment (sadiabbasi) [reply test-integration/externalUser.test.ts:218]: ohho let me remove this change from this change as i added this on username display task

File: src/sdk/controllers/userApps.ts [hunk 4]
```diff
@@ -102,7 +102,12 @@ export class UserApps {
      *
      * @returns the requests (JWTs), username and accountName
      */
-    static getLoginRequestParams(): { requests: string; username: string; accountName: string } {
+    static getLoginRequestParams(): {
+        requests: string;
+        username: string;
+        accountName: string;
+        response: { success: boolean; reason: SdkErrors };
+        } {
```
Comment (Rebal67) [inline src/sdk/controllers/userApps.ts:110]: can you change that to type because it became big 
also update the TSDocs

**Pull request merged**